node/src/actors/chain_manager/mod.rs

```diff
@@ -1735,6 +1749,111 @@ impl ChainManager {
+                inventory_manager
+                    .send(GetItemBlock { hash })
+                    .then(move |res| match res {
+                        Ok(Ok(block)) => {
+                            let mut transactions = vec![];
+
+                            // TODO: be functional
+                            for vtt in &block.txns.value_transfer_txns {
+                                transactions.push(Transaction::ValueTransfer(vtt.clone()));
+                            }
+                            for drt in &block.txns.data_request_txns {
+                                transactions.push(Transaction::DataRequest(drt.clone()));
+                            }
+                            // TODO: only include reveals of data requests that were in reveal
+                            // stage right at the revert epoch. Reveals that point to data requests
+                            // that are still in commit stage will not be valid, and reveals that
+                            // point to data requests that did not exist at the revert epoch will
+                            // also not be valid.
+                            for reveal in &block.txns.reveal_txns {
+                                transactions.push(Transaction::Reveal(reveal.clone()));
+                            }
+                            futures::future::ok(transactions)
+                        }
+                        Ok(Err(e)) => {
+                            log::error!("Error in GetItemBlock {}: {}", hash, e);
+                            futures::future::err(())
+                        }
+                        Err(e) => {
+                            log::error!("Error in GetItemBlock {}: {}", hash, e);
+                            futures::future::err(())
+                        }
+                    })
+                    // TODO: make sure that we want to ignore errors
+                    .then(|x| futures::future::ok(x.ok()))
+            });
+
+            join_all(aux)
+                // Map Option<Vec<Vec<T>>> to Vec<T>, this returns all the non-error results
+                .map(|x| {
+                    x.into_iter()
+                        .flatten()
+                        .flatten()
+                        .collect::<Vec<Transaction>>()
+                })
+        })
+        .into_actor(self)
+        .and_then(move |transactions, act, ctx| {
+            log::debug!(
+                "{} transactions since epoch {}: {:?}",
+                transactions.len(),
+                epoch,
+                transactions
+            );
+
+            // FIXME(#1763): Do not use a run_later
+            let five_epochs = u64::from(
+                act.consensus_constants()
+                    .checkpoints_period
+                    .saturating_mul(5),
+            );
+            ctx.run_later(Duration::from_secs(five_epochs), move |_act, ctx| {
+                mempool_transactions.extend(transactions);
```

Comment:
Let's move the debug log here:

```suggestion
                mempool_transactions.extend(transactions);
                log::debug!(
                    "Re-adding {} transactions into mempool since epoch {}",
                    mempool_transactions.len(),
                    epoch,
                );
```